stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -0,0 +1,367 @@
+        # delivery route will use them.
+        if not self.out_type_id:
+            self.out_type_id = self.warehouse_id.out_type_id.copy(
+                {
+                    "name": (
+                        f"{self.warehouse_id.out_type_id.name} "
+                        f"{self.sequence_prefix}"
+                    ),
+                    "active": True,
+                }
+            )
+            self.out_type_id.sequence_id.prefix += f"{self.sequence_prefix}/"
+        if "pick_" in self.delivery_steps:
+            if not self.pick_type_id:
+                self.pick_type_id = self.warehouse_id.pick_type_id.copy(
+                    {
+                        "name": (
+                            f"{self.warehouse_id.pick_type_id.name} "
+                            f"{self.sequence_prefix}"
+                        ),
+                        "active": True,
+                    }
+                )
+                self.pick_type_id.sequence_id.prefix += f"{self.sequence_prefix}/"
+            # Create a dedicated 'Output/SUB' location
+            if not self.output_stock_loc_id:
+                self.output_stock_loc_id = (
+                    self.warehouse_id.wh_output_stock_loc_id.copy(
+                        {
+                            "name": self.sequence_prefix,
+                            "location_id": self.warehouse_id.wh_output_stock_loc_id.id,
+                        }
+                    )
+                )
+                self.pick_type_id.default_location_dest_id = self.output_stock_loc_id
+                self.out_type_id.default_location_src_id = self.output_stock_loc_id
+            if "pack_" in self.delivery_steps:
+                if not self.pack_type_id:
+                    self.pack_type_id = self.warehouse_id.pack_type_id.copy(
+                        {
+                            "name": (
+                                f"{self.warehouse_id.pack_type_id.name} "
+                                f"{self.sequence_prefix}"
+                            ),
+                            "active": True,
+                        }
+                    )
+                    self.pack_type_id.sequence_id.prefix += f"{self.sequence_prefix}/"
+                # Create a dedicated 'Packing Zone/SUB' location
+                self.pack_stock_loc_id = self.warehouse_id.wh_pack_stock_loc_id.copy(
+                    {
+                        "name": self.sequence_prefix,
+                        "location_id": self.warehouse_id.wh_pack_stock_loc_id.id,
+                    }
+                )
+                self.pick_type_id.default_location_dest_id = self.pack_stock_loc_id
+                self.pack_type_id.default_location_src_id = self.pack_stock_loc_id
+                self.pack_type_id.default_location_dest_id = self.output_stock_loc_id
+        # Enable relevant locations (pack/output) based on the delivery steps
+        self.warehouse_id._update_location_delivery(self.delivery_steps)
```

Comment:
The warehouse locations should not be modified as the flow has its own intermediate locations
```suggestion
```